### PR TITLE
haddock bundle script

### PR DESCRIPTION
**Pull request by phadej**

### CODE REVIEW

**Comment by adamgundry** [comment]
Nice! Thanks for looking into this. I haven't looked extremely closely at the code, but the results on https://hackage.haskell.org/package/optics look good. :smile: 

Did you manage to resolve the issue with there being a few missing links?

We should remove the note from the description saying "Hackage does not yet display documentation for reexported-modules".

Fixes #237.

**Comment by arybczak** [comment]
It seems that modules from `optics-extra` are missing links :disappointed:

**Comment by arybczak** [comment]
I wonder if providing dummy files that redirect to modules in appropriate package with `<meta http-equiv="refresh" content="0; url=...>` would work :thinking:

**Comment by phadej** [comment]
> Did you manage to resolve the issue with there being a few missing links?

No, that seems to be some issue with `haddock`, i have no idea what it could be.

File: haddockbundle.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#!/bin/sh
+
```
Comment (arybczak) [inline haddockbundle.sh:2]: Can you add a brief "Usage" section? Even if it's running it without any arguments.

**Comment by phadej** [comment]
I added a commit which seems to fix links from `optics-extra` packages

**Comment by adamgundry** [comment]
Nice work tracking that down! And sorry for letting this linger, let's get it merged. I'm assuming you already uploaded docs to Hackage, since I see them published...

I think we should:
 * Remove the note from the description saying "Hackage does not yet display documentation for reexported-modules".
 * Briefly document the release process somewhere. Is the only new step is running `haddockbundle.sh` and uploading the generated `dist-newstyle/optics-bundled-docs.tar`, and can one do that directly with `cabal upload -d`?

**Comment by phadej** [comment]
>  can one do that directly with cabal upload -d

I don't know, never used that one.

> Briefly document the release process somewhere.

Once someone does it, they could write a trace of what they did.

**Review by adamgundry — APPROVED** [review]

**Pull request merged**